components/experimental/src/units/converter_factory.rs

```diff
@@ -38,20 +38,55 @@ impl From<Sign> for num_bigint::Sign {
     }
 }
 
-impl<'data> ConverterFactory<'data> {
-    #[cfg(feature = "datagen")]
-    pub fn from_payload(
-        payload: &'data UnitsInfoV1<'data>,
-        payload_store: &'data ZeroTrieSimpleAscii<ZeroVec<'data, u8>>,
-    ) -> Self {
-        Self {
-            payload,
-            payload_store,
-        }
+impl ConverterFactory {
+    icu_provider::gen_any_buffer_data_constructors!(
+        locale: skip,
+        options: skip,
+        error: ConversionError,
```

Comment:
Return `DataError` here (new policy decided in https://github.com/unicode-org/icu4x/pull/4638#issuecomment-1994218244)

Reply:
Done